### PR TITLE
ci(feat): Add Docker compose building workflow

### PR DESCRIPTION
**Pull request by armenzg**
This ensures that we can still build the Docker images.

There's also few minor changes.

### CODE REVIEW

File: backend-py/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
-# Installing packages...
+
 WORKDIR /backend-py
-COPY . .
+# Re-create the requirements layer if the requirements change
+COPY requirements.txt .
 RUN pip install -r requirements.txt
+
+# Copy the rest of the source code now, otherwise, code changes will invalidate the requirements cache
+COPY ./src/* /backend-py/
```
Comment (armenzg) [inline backend-py/Dockerfile:9]: We don't need all the files under backend-py but only the source files.

File: backend-py/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,11 @@
 FROM python:3.9 as backend-py
```
Comment (asottile-sentry) [inline backend-py/Dockerfile:1]: probably want to use one of the slim variants rather than the bulky i mage

**Review by asottile-sentry — APPROVED** [review]

File: backend-py/Dockerfile (hunk 2 above)
Comment (armenzg) [reply backend-py/Dockerfile:1]: I left it out of this PR to reduce changes.

**Pull request merged**